src/timetable.js

```diff
@@ -0,0 +1,133 @@
+import R from 'ramda'
+import moment from 'moment'
+import { setDateToZeroTime, weekdayNum } from './date'
+
+export function createTimeline (grid) {
+  return {
+    start: grid.starts * 3600,
+    end: grid.ends * 3600,
+    duration: (grid.ends - grid.starts) * 3600,
+    hourDuration: grid.lessonDuration * 3600,
+    hours: (grid.ends - grid.starts) / grid.lessonDuration,
+    firstHour: grid.facultyGrid ? 1 : Math.ceil(grid.starts),
+    offset: grid.facultyGrid ? 0 : grid.starts % 1,
+  }
+}
+
+export function calculateEventPosition (event, timeline) {
+  const eventStartDate = new Date(event.startsAt)
+  const eventStartTs = eventStartDate.getTime() / 1000
+  const eventEndTs = new Date(event.endsAt).getTime() / 1000
+  const dayStartTs = setDateToZeroTime(eventStartDate).getTime() / 1000
+  const timelineStartTs = dayStartTs + timeline.start
+
+  return {
+    position: (eventStartTs - timelineStartTs) / timeline.duration,
+    length: (eventEndTs - eventStartTs) / timeline.duration,
+  }
+}
+
+export function calculateHourLabels (timeline) {
+  return R.times((n) => {
+    return {
+      id: 'hlbl-' + n,
+      label: timeline.firstHour + n,
+      position: (n + timeline.offset) / timeline.hours,
+      length: (1 / timeline.hours),
+    }
+  }, Math.ceil(timeline.hours))
+}
+
+export function mapPropertiesToClass (properties, elementClass) {
+  let className = elementClass + ' '
+
+  for (let property in properties) {
```

Comment:
Use [R.reduce](http://ramdajs.com/0.18.0/docs/#reduce) etc.

Note for myself: refactor this function.